Sources/Plasma/PubUtilLib/plAudio/plAudioSystem.cpp

```diff
@@ -670,13 +598,14 @@ void    plAudioSystem::IUpdateSoftSounds(const hsPoint3 &newPosition)
     fDebugActiveSoundDisplay->AddLine(0xff00ffff, "Incidentals");
     fDebugActiveSoundDisplay->AddLine("--------------------");
 
-    for (i = 0; sortedList && i < fMaxNumSounds; sortedList = sortedList->fSortNext) {
-        plSound* sound = plSound::ConvertNoRef(sortedList->fSoundKey->GetObjectPtr());
+    auto soundIt = fActiveSofts.begin();
+    for (i = 0; soundIt != fActiveSofts.end() && i < fMaxNumSounds; soundIt++) {
+        plSound* sound = (plSound*)soundIt->fSoundKey->GetObjectPtr();
```

Comment:
Safe to cast directly instead of ConvertNoRef?

Reply:
Yes, only `plSound`s ever get added to this collection. The loop is tight enough that I feel it's worth cutting out as much code as possible.